# エラー対処法いろいろ

## ダイアログ：Dockerからエラーが返されました。Dockerデーモンが実行されていることを確認します
### 解決方法：Dockerを起動する
特徴：コンテナで開くを選択した直後、読み込みを挟まずに出る

## コンテナを開く処理中にエラーが返される
### だいたいこれすりゃ直る！
1. MySQLの実行を停止する
    1. タスクマネージャーから「MySQL.exe」的な名前のタスクを終了させる
    2. リンク先のやつもするとなお良し https://qiita.com/Shime_texas/items/03b49cf42fbdaf0e2250
2. リビルドする
    1. 左下の><みたいなマークをクリック
    2. 「コンテナーのリビルド」を選択

## 「python manage.py」実行時エラーログ：...[Errno 2] No such file or directory
1. カレントディレクトリ要チェック。manage.pyが存在するディレクトリで実行
2. (必要であれば)pwd、cdコマンドを活用し、manage.pyが存在するディレクトリに移動
3. 再度「python manage.py」コマンドを実行

## 「python manage.py runserver 0.0.0.0:8000」実行時エラーログ：django.core.exceptions.ImproperlyConfigured: Set the SECRET_KEY environment variable
1. .envファイルが無い。appディレクトリが存在する階層に追加してやること

## クローン、コミット時などのエラーダイアログ：Gitの"user.name"と"user.email"を構成していることを確認してください
### 解決方法：ユーザー情報を入力する
1. Git Bashでコマンド実行：git config user.name　→　自身のユーザー名を入力
2. Git Bashでコマンド実行：git config user.email　→　自身のメールアドレスを入力
3. Git Bashでコマンド実行：git config --global --list ... 自身のユーザー名、メールアドレスが出力されます

## おまけ：runserverした後の http://0.0.0.0:8000/ 踏んでもこのサイトにアクセスできませんって言われるんですけど
1. http://127.0.0.1:8000/ です。.envファイルに記載あり ALLOWED_HOSTS=localhost,127.0.0.1 

### あとはREADME通りにやると何とかなります